### ISSUE TITLE
RequestParameters error AWS::ApiGateway::Method

### ISSUE THREAD

**Issue by mcgoosh** [aws-cloudformation/cfn-python-lint#1412]
cfn-lint 0.28.1

When using [`AWS::ApiGateway::Method`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-method.html) and specifically [`RequestParameters`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-method.html#cfn-apigateway-method-requestparameters) the linter is giving me an error in my editor.  Even though the template works.  

`      RequestParameters:
          method.request.querystring.product_name: "'false'"`

> [cfn-lint] E3012: Property Resources/ApiGatewayMethodGET1/Properties/RequestParameters/method.request.querystring.product_name should be of type Boolean

If I do not put the boolean in quotes the cfn-linter doesn't give me errors but the template does not work and errors with 

> Value of property RequestParameters must be an object

**Comment by kddejong** [comment]
I'm trying to repeat this issue but I'm not getting an error with this resource.  
```
Method:
    Type: AWS::ApiGateway::Method
    Properties:
      HttpMethod: POST
      ResourceId: !GetAtt RestApi.RootResourceId
      RestApiId: !Ref RestApi
      AuthorizationType: NONE
      Integration:
        Type: MOCK
        ContentHandling: !Ref contentHandling
        IntegrationResponses:
          - ContentHandling: !Ref contentHandling
            StatusCode: "400"
      RequestParameters:
        method.request.querystring.product_name: false
      OperationName: !Ref operationName
```

We do strong type checking based on the resource spec.  You can disable it by following the documentation [here](https://github.com/aws-cloudformation/cfn-python-lint#configure-rules) `E3012:strict=false`.  In this configuration we will still evaluate if the string is a boolean.
```
"RequestParameters": {
          "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-method.html#cfn-apigateway-method-requestparameters",
          "DuplicatesAllowed": false,
          "PrimitiveItemType": "Boolean",
          "Required": false,
          "Type": "Map",
          "UpdateType": "Mutable"
        },
```

**Comment by mcgoosh** [comment]
Ugh sorry for wasting your time Kevin.  It does indeed work after fixing a white space problem.  Thanks for looking at it.
